webapp/api/sso.py

```diff
@@ -0,0 +1,47 @@
+import os
+from webapp import api
+from webapp.api.exceptions import ApiResponseError, ApiResponseDecodeError
+
+api_session = api.requests.Session(timeout=(.5, 6))
+
+DASHBOARD_API = os.getenv(
+    "DASHBOARD_API", "https://dashboard.snapcraft.io/dev/api/"
+)
+
+LOGIN_URL = os.getenv("LOGIN_URL", "https://login.ubuntu.com")
+
+
+HEADERS = {
+    "Accept": "application/json, application/hal+json",
+    "Content-Type": "application/json",
+    "Cache-Control": "no-cache",
+}
+
+
+def process_response(response):
+    try:
+        body = response.json()
+    except ValueError as decode_error:
+        api_error_exception = ApiResponseDecodeError(
+            "JSON decoding failed: {}".format(decode_error)
+        )
+        raise api_error_exception
+
+    if not response.ok:
```

Comment:
It's usually best to check that the response code is what you expect before you try to decode the response body, or you're going to get an uninformative JSON decode failure when you get a non-JSON error, and you'll have no idea what the response code was. response.ok is also probably not what you want, since that is true if the response is a redirect, for example.

Reply:
makes sense
Done